### DISCUSSION TITLE
Inconsistent behaviour v6.14.x >< v7.4 and later

### DISCUSSION THREAD

**Discussion by MtthsB** [react-hook-form/react-hook-form#5139] [Q&A]
**Describe the bug**
We recently tried to move up a major RHF version (from v6.14 to v7.4), and discovered something that might be a bug, an accidental oversight on the part of the developer team, or even an intended change (in which case some additional documentation might be in order).

The behaviour in question arises when passing in defaultValues into a `useForm` hook, which we use for editing information, pre-filling the form fields with the existing data. In v6.14, even though the incoming data (which is passed along as the form's default values) might include form-unrelated information (such as database-related timestamps or ids), when submitting the (edited) form, only the editable fields would be included in the submitted data.

However, in v7.4 (and v7.5 based on my tests), all information passed into the form as its default values, regardless of their editable nature, will be included in the submission data.

This leads to validation errors in the backend.

**To Reproduce**

1. Go to https://codesandbox.io/s/rhf-test-ysm24
2. Submit the form.
3. Notice how the submitted data is equal to the form's default values
4. Switch RHF's version to v6.14.2
5. Adapt the Controller's render prop function arguments to match v6.14.2 signature in order to make the code work.
6. Submit the form.
7. Notice the difference in incoming and outgoing information.

**Codesandbox link (Required)**
https://codesandbox.io/s/rhf-test-ysm24

**Desktop (please complete the following information):**

- OS: macOS Big Sur 11.2.3
- Browser: Firefox 
- Version: 87.0

**Comment by mbretter** [comment]
you have to set shouldUnregister to true, here is the updated CSB:
https://codesandbox.io/s/rhf-test-forked-ym84j?file=/src/FormProvider.jsx
